### ISSUE TITLE
Move away from np.random.RandomStates

### ISSUE THREAD

**Issue by AlMrvn** [quantumlib/Cirq#6321]
**Is your feature request related to a use case or problem? Please describe.**
the [random generator of cirq](https://github.com/quantumlib/Cirq/blob/96b38428a8b67e469f8ea01283c3cd8502bbf7df/cirq-core/cirq/value/random_state.py#L40) rely on [legacy code from numpy](https://numpy.org/doc/stable/reference/random/legacy.html): 

**Describe the solution you'd like**
use the "new numpy way" of dealing with [random number generators](https://numpy.org/doc/stable/reference/random/generator.html#numpy.random.default_rng): 

**What is the urgency from your perspective for this issue? Is it blocking important work?**
P3 - I'm not really blocked by it, it is an idea I'd like to discuss / suggestion based on principle

**Comment by mpharrigan** [comment]
Yes, I put a lot of the `RandomState`s there in the first place. The new way (last I checked) has a bit more boilerplate but is better designed. The general philosophy of using an actual random generator object everywhere we can instead of the `np.random` module global-generator is still a good one

**Comment by arettig** [comment]
Closing this issue in favor of the more detailed/updated #8374.
